app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/ChatOpenFragment.kt

```diff
@@ -93,10 +96,34 @@ class ChatOpenFragment : Fragment(R.layout.fragment_chat) {
         }
 
         sendRunButton.setOnClickListener {
-            // Handle sending run path message here
+            // Observe the LiveData object
+            userModelCached.getRunHistory().observe(viewLifecycleOwner) { runHistory ->
+                val runPopupMenu = PopupMenu(requireContext(), sendRunButton)
+                runHistory.forEach { run ->
+                    runPopupMenu.menu.add(run.getDate())
+                }
+                runPopupMenu.setOnMenuItemClickListener { menuItem ->
+                    val selectedRunTitle = menuItem.title.toString()
+                    // Find the selected run in the user's run history
+                    val selectedRun =
+                        runHistory.firstOrNull { it.getDate() == selectedRunTitle }
+                    if (selectedRun != null) {
+                        // If a run is selected, create a new Message
+                        val newMessage = Message.createRunPathMessage(
+                            userId,
+                            selectedRun,
+                            System.currentTimeMillis(),
+                        ) // replace with appropriate method to create run path message
+                        messagesList.add(newMessage)
+                        messagesAdapter.notifyDataSetChanged()
+                        database.addChatMessage(chatId, newMessage)
+                    }
+                    true
+                }
+                runPopupMenu.show()

```

Comment:
We probably want this code to be moved outside of the `onViewCreated` method.

Reply:
Done, I created a helper function for it.